### PR TITLE
Removed `ReflectionMethod::getAst()`, `getBodyAst()`, `getBodyCode()` and `getReturnStatementsAst()` to prevent memory leaks

### PR DESCRIPTION
**Pull request by kukulich**
Part of https://github.com/Roave/BetterReflection/issues/1073

### CODE REVIEW

File: src/Reflection/ReflectionFunctionAbstract.php [hunk 1]
```diff
@@ -28,13 +26,107 @@
...
+        $this->parameters       = $this->createParameters($node);
+        $this->returnsReference = $node->returnsByRef();
+        $this->returnType       = $this->createReturnType($node);
+        $this->attributes       = ReflectionAttributeHelper::createAttributes($this->reflector, $this, $node->attrGroups);
+        $this->docComment       = GetLastDocComment::forNode($node);
+        $this->couldThrow       = $this->computeCouldThrow($node);
+
+        if ($node->hasAttribute('startLine')) {
+            $startLine = $node->getStartLine();
+            assert($startLine > 0);
```
Comment (Ocramius) [inline src/Reflection/ReflectionFunctionAbstract.php:78]: Hmm, weird mutation: incompatible with type declaration.

I wonder if it's uncovered code not handled by pcov/phpdbg 🤔

File: src/Reflection/ReflectionFunctionAbstract.php [hunk 2]
```diff
@@ -213,10 +284,21 @@ public function isVariadic(): bool
...
     {
+        return $this->couldThrow;
+    }
+
+    private function computeCouldThrow(MethodNode|Node\Stmt\Function_|Node\Expr\Closure|Node\Expr\ArrowFunction $node): bool
+    {
+        $statements = $node->getStmts();
+
+        if ($statements === null) {
+            return false;
```
Comment (Ocramius) [inline src/Reflection/ReflectionFunctionAbstract.php:293]: A test with an empty method may be missing? 🤔

File: src/Reflection/ReflectionType.php [hunk 3]
```diff
@@ -69,6 +69,9 @@ public static function createFromNode(
         return new ReflectionUnionType($reflector, $owner, new UnionType($types));
     }
 
+    /** @internal */
+    abstract public function withOwner(ReflectionParameter|ReflectionMethod|ReflectionFunction|ReflectionEnum|ReflectionProperty $owner): static;
```
Comment (Ocramius) [inline src/Reflection/ReflectionType.php]: Not sure this should be interfaced.

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: src/Reflection/ReflectionFunctionAbstract.php (hunk 2 above)
Comment (kukulich) [reply src/Reflection/ReflectionFunctionAbstract.php:293]: Yes, added.

File: src/Reflection/ReflectionType.php (hunk 3 above)
Comment (kukulich) [reply src/Reflection/ReflectionType.php]: Removed.

File: src/Reflection/ReflectionFunctionAbstract.php (hunk 1 above)
Comment (kukulich) [reply src/Reflection/ReflectionFunctionAbstract.php:78]: It's strange because the some code in other reflections is OK.

**Review by Ocramius — APPROVED** [review]
👍 🚢 

Well done!

**Pull request merged**